### ISSUE TITLE
Incorrect type for ws.publish

### ISSUE THREAD

**Issue by klavinski** [elysiajs/elysia#830]
### What version of Elysia is running?

1.1.12

### What platform is your computer?

Darwin 24.1.0 arm64 arm

### What steps can reproduce the bug?

In an `Elysia.ws` handler, add a `response` validator and use `ws.publish`, like this:

```typescript
new Elysia()
    .ws( "/", {
        open: ws => {
            ws.publish( "channel", "hello" )
        },
        response: t.String()
    } )
    .listen( 3000 )
```

### What is the expected behavior?

There should be no type error.

### What do you see instead?

`"hello"` has a type error:

> Argument of type `string` is not assignable to parameter of type `{ 200: string; }`.

### Additional information

Essentially, when `response` validates the type `T`, the type checking of `data` argument requires `{200: T}`. However, the execution expects `T`: if provided with a `{200: T}`, a validation error is thrown.

I believe a [previous version](https://github.com/elysiajs/elysia/blob/541043102bc43ea7768f8983bd43dbb909affb72/src/ws/index.ts#L35) had the correct type for the `data` argument: `WS['data']['schema']['response']` instead of `Route["response"]`.

### Have you try removing the `node_modules` and `bun.lockb` and try again yet?

No.

**Comment by klavinski** [comment]
The generated type for Eden Treaty's `subscribe` is also wrong. I currently use it like this:

```typescript
treatyClient.websocketRoute.subscribe().subscribe( ( { data } ) => {
    const wellTypeData = data as unknown as ( typeof data )[ 200 ] )
} )
```

**Comment by SaltyAom** [comment]
Fixed and published under [1.1.13](https://github.com/elysiajs/elysia/releases/tag/1.1.13) with test case included
see https://github.com/elysiajs/elysia/compare/1.1.12...1.1.13

If this doesn't fix the issue, feel free to tag me or reopen the issue.
